src/transformers/trainer.py

```diff
@@ -3876,21 +3895,15 @@ def evaluation_loop(
         if num_samples == 0 and observed_num_examples > 0:
             num_samples = observed_num_examples
 
-        # Metrics!
+        # Call compute metrics on epoch end only if we didn't compute them on each batch.
         if (
-            self.compute_metrics is not None
+            not args.batch_eval_metrics
+            and self.compute_metrics is not None
             and all_preds is not None
             and all_labels is not None
-            and not self.args.batch_eval_metrics
         ):
-            if args.include_inputs_for_metrics:
```

Comment:
We don't need to check it, if `args.include_inputs_for_metrics=False` then `all_inputs` is going to be `None`